test/builder/vcsim_test_context.go

```diff
@@ -144,6 +147,9 @@ type TestContextForVCSim struct {
 	datastore *object.Datastore
 
 	azCCRs map[string][]*object.ClusterComputeResource
+
+	// WithWorkloadIsolation stores VCSimTestConfig.WithWorkloadIsolation value.
```

Comment:
I don't think this needs to be public. Also group this near the existing `WithFaultDomains ...` comment (and maybe update that comment a bit).